csaf_2.0/prose/csaf-v2-editor-draft.md

```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
+Any cryptographic hash object has the 3 mandatory properties `algorithm`, `file`, and `value`.
+
+The algorithm of the cryptographic hash representation (`algorithm`) of type `string` with one or more characters contains the name of the cryptographic hash algorithm used to calculate the value.
+The default value for `algorithm` is `SHA-3`.
 
 Examples:
 
-    Microsoft Host Integration Server 2006 Service Pack 1
-    Cisco AnyConnect Secure Mobility Client 2.3.185
+    SHA-256
+    SHA-384
+    SHA-512
+    SHA-3
+    BLAKE3
 
-#### 3.1.3.3 Full Product Name Type - Product ID
+The file representation (`file`) of type `string` with one or more characters contains the name of the file which is identified by the hash value.
 
-Product ID (`product_id`) holds a value of type Product ID (`product_id_t`).
+Examples:
+
+    WINWORD.EXE
+    msotadddin.dll
+    sudoers.so
+
+The Value of the cryptographic hash representation (`value`) of value type `string` of 64 or more characters with `pattern` (regular expression):
+
+    ^[0-9a-fA-F]{64,}$
+
+The Value of the cryptographic hash attribute ontains the cryptographic hash value.
+
+Examples:
+
+    4775203615d9534a8bfca96a93dc8b461a489f69124a130d786b42204f3341cc
+    9ea4c8200113d49d26505da0e02e2f49055dc078d1ad7a419b32e291c7afebbb84badfbd46dec42883bea0b2a1fa697c
+    37df33cb7464da5c7f077f4d56a32bc84987ec1d85b234537c1c1a4d4fc8d09dc29e2e762cb5203677bf849a2855a0283710f1f5fe1d6ce8d5ac85c645d0fcb3
+
+##### 3.1.3.3.3 Full Product Name Type - Product Identification Helper - PURL
+
+The package URL (PURL) representation (`purl`) is a `string` of 4 or more characters with `pattern` (regular expression):
+
+    ^pkg:
+
+This package URL (PURL) attribute refers to a method for reliably identifying and locating software packages external to this specification.
+
+##### 3.1.3.3.4 Full Product Name Type - Product Identification Helper - Serial Numbers
+
+The list of serial numbers representation (`serial_numbers`) of value type `array` with 1 or more items contains a list of parts, or a full serial numbers.
+
+A list of serial numbers SHOULD only be used if a certain range of serial numbers with a software version is affected, or the serial numbers change during update.
+
+Any given serial number of value type `string` with at least 1 character represents a part, or a full serial number of the component to identify.
+
+    "serial_numbers": {
+        //...  
+      "items": {
+        //...  
+      }
+    }
+
+
+##### 3.1.3.3.5 Full Product Name Type - Product Identification Helper - Generic URIs
+
+List of generic URIs representation (`x_generic_uris`) of value type `array` with at least 1 item contains a list of identifiers which are either vendor-specific or derived from a standard not yet supported.
+
+Any such Generic URI item of value type `object` provides the two mandatory properties `namespace` and URI (`uri`).
```

Comment:
I would replace `namespace` with Namespace (`namespace`) or remove the word URI (and the brackets).

Reply:
Resolved with 7512432.